compiler-access/src/main/scala/org/alephium/ralph/lsp/access/compiler/InconsistentWarningsFix.scala

```diff
@@ -0,0 +1,244 @@
+// Copyright 2024 The Alephium Authors
+// This file is part of the alephium project.
+//
+// The library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the library. If not, see http://www.gnu.org/licenses/.
+
+package org.alephium.ralph.lsp.access.compiler
+
+import org.alephium.ralph._
+import org.alephium.protocol.vm._
+
+private object InconsistentWarningsFix {
+
+  /*
+   * The mutable state of `Ast.ContractWithState` is causing inconsistent warnings when compiling twice the same AST.
+   * This function fixes the issue by performing a deep copy of the `Ast.ContractWithState` objects.
+   * So the AST in `Tree.Source` is not mutated and we compile the copy of the ast.
+   *
+   * FIXME:
+   * For some reason we still need to call `genStatefulContracts` to have all tests passing, some Go-To-Definition tests are failing.
+   * But `genStatefulScripts` should not be called.
+   */
+  def fix(
+      contracts: Seq[Ast.ContractWithState],
+      structs: Seq[Ast.Struct],
+      options: CompilerOptions): Seq[Ast.ContractWithState] = {
+    // Mutate just what we need to have all go-to-definition working
+    // TODO: We could remove this later if we find a better solution
+    Ast.MultiContract(contracts, structs, None, None).extendedContracts().genStatefulContracts()(options)
+
+    // Deep copy the AST to avoid inconsistent warnings
+    contracts.map(copyContractWithState)
```

Comment:
The go-to def and code-completion test-cases fail because somewhere in the copy the type information [`tpe`](https://github.com/alephium/alephium/blob/5c89bcde6cb007073ed96a37a87afa7971ea3443/ralph/src/main/scala/org/alephium/ralph/Ast.scala#L184) is lost.

Reply:
>The go-to def and code-completion test-cases fail because somewhere around the copy the type information [tpe](https://github.com/alephium/alephium/blob/5c89bcde6cb007073ed96a37a87afa7971ea3443/ralph/src/main/scala/org/alephium/ralph/Ast.scala#L184) is lost 

nice catch, let me try to add it back